interfaces/custom_modal_interface.py

```diff
@@ -0,0 +1,7 @@
+import abc
+
+
+class CustomModalInterface(metaclass=abc.ABCMeta):
```

Comment:
I'm not sure the two abstract class in `interfaces/` have much utility. What is the idea you had in mind when creating them?